### PR TITLE
Change primer/behaviors build script to support esm and cjs

### PR DESCRIPTION
**Pull request by pksjce**
This is an attempt to fix https://github.com/primer/react/issues/1762

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 node_modules/
 dist/
```
Comment (dgreif) [inline .gitignore:2]: Can remove `dist` here and in package.json since everything is in `lib` now

File: package.json [hunk 2]
```diff
@@ -2,25 +2,41 @@
...
+    "utils",
+    "lib",
+    "lib-esm",
+    "lib/utils"
   ],
   "sideEffects": [
-    "dist/focus-zone.js",
-    "dist/focus-trap.js"
+    "lib/focus-zone.js",
+    "lib/focus-trap.js"
```
Comment (dgreif) [inline package.json:28]: We should add the `lib-esm` versions of these files as well

**Review by dgreif — COMMENTED** [review]
It looks like you took the same approach here as what we are doing in PRC.  While that should work, I think it's a bit more complicated than we actually need in this case.  We are currently using `tsc` to compile the typescript files, which supports both cjs and esm as an output.  I put together an alternate approach in https://github.com/primer/behaviors/commit/caf98b36d73952364ded2466ba272a839183bf03 but haven't tested it out with npm linking, etc.  Let me know what you think!

**Comment by pksjce** [comment]
> It looks like you took the same approach here as what we are doing in PRC. While that should work, I think it's a bit more complicated than we actually need in this case. We are currently using `tsc` to compile the typescript files, which supports both cjs and esm as an output. I put together an alternate approach in [caf98b3](https://github.com/primer/behaviors/commit/caf98b36d73952364ded2466ba272a839183bf03) but haven't tested it out with npm linking, etc. Let me know what you think!

I love this! I didn't guess `tsc` could do all that work for me!
I just tested your branch on my PR here and put that in my nextjs project and it works! Do you want to make a PR for this?

One thing I wanted to ask was, is there anyway for us to get rid of `utils/package.json` ?

**Comment by dgreif** [comment]
@pksjce I've opened #42.  

> One thing I wanted to ask was, is there anyway for us to get rid of `utils/package.json` ?

This extra package.json is needed for consumers that use TypeScript.  This allows them to do `import {things} from '@primer/behaviors/utils'`.  While most bundlers (webpack, etc) and node will respect the `exports` field in the top-level package.json, TypeScript does not yet have support for defining submodules in this way.  Creating this extra `utils/package.json` is a way of _tricking_ TypeScript so that it recognizes the submodule.  Hopefully they get this feature added to TypeScript soon, but until then we need to keep it around.

**Comment by pksjce** [comment]
Yeah, I could find no way of removing it either. I am following https://github.com/microsoft/TypeScript/issues/33079 to see if they ever fix this. At that time, we can just kill that `package.json` file. Thanks for confirming this. We have a similar typescript problem with @primer/react/drafts. I might use this hack as a fix there too.

Thanks for the PR. I will close this one in reference to https://github.com/primer/behaviors/pull/42

**Pull request closed without merging**